Source/Request.swift

```diff
@@ -1282,17 +1286,29 @@ public final class DataStreamRequest: Request {
         appendResponseSerializer {
             self.underlyingQueue.async {
                 self.responseSerializerDidComplete {
-                    queue.async {
-                        do {
-                            let completion = Completion(request: self.request,
-                                                        response: self.response,
-                                                        metrics: self.metrics,
-                                                        error: self.error)
-                            try stream(.init(event: .complete(completion), token: .init(self)))
-                        } catch {
-                            // Ignore error, as errors on Completion can't be handled anyway.
-                        }
-                    }
+                    self.enqueueCompletion(on: queue, stream: stream)
+                }
+            }
+        }
+    }
+
+    func enqueueCompletion<Success, Failure>(on queue: DispatchQueue,
+                                             stream: @escaping Handler<Success, Failure>) {
+        $streamMutableState.read { state in
+            guard state.numberOfExecutingStreams == 0 else {
+                underlyingQueue.async { self.enqueueCompletion(on: queue, stream: stream) }
```

Comment:
Would you want a delay here at all? Seems like you're just re-enqueuing as fast as possible which may not be ideal.

Reply:
You're right, this will requeue rather quickly. The check is cheap, but I'm going to investigate a better solution.

Reply:
Okay, I've implemented an alternate solution that stores the delayed completion until all streams have finished. Check it out in https://github.com/Alamofire/Alamofire/pull/3216/commits/9a72a35016ef164f9fcbcb20a8b436a6bb388914.